SearchableToolsTests/SearchableToolsTests.swift

```diff
@@ -50,7 +50,11 @@ final class SearchableToolsTests: XCTestCase {
         
         viewModel.addItemName = "Prueba"
         
-        let lastItem = viewModel.filteredData.count
+        viewModel.addButton()
+        
+        var lastItem = viewModel.filteredData.count
+        
+        lastItem = lastItem - 1
         
         XCTAssertEqual(viewModel.filteredData[lastItem], viewModel.addedItem)
```

Comment:
You shouldn't be looking into the `filteredData` array because you are not adding the new tool into the filtered tools, the new tool is being added to the `tools` array `tools.append(addItemName)`.
```suggestion
        XCTAssertEqual(viewModel.tools[lastItem], "Prueba")
```